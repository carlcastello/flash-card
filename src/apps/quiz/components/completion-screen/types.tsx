export interface IOwnProps {
  onClose: () => void,
  onReview: () => void,
  classes: {
    
  }
}