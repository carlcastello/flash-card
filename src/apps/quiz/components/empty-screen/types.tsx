export interface IOwnProps {
  onEdit: () => void,
  onClose: () => void,
  classes: {

  }
}