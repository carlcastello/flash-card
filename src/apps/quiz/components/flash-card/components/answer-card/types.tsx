import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  answer: string,
  flashcardStatus: FlashcardStatus,
  ref: any,
  classes: {
    paper: string,
    input: string,
    typography: string
  }
}

export interface IOwnState {
  answer: string,
  hideAnswer: boolean,
}