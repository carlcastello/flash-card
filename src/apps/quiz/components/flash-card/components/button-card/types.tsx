import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  flashcardStatus: FlashcardStatus,
  submit: () => void,
  next: () => void,
  classes: {
    button: string
  }
}

export interface IOwnState {

}