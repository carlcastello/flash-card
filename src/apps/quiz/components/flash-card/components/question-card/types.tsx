import { FlashCardStatus } from "../../types";
import { QuestionType } from "../../../../../commons/types";

export interface IQuestionObject {
  question: string,
  subQuestion?: string,
  hint?: string,
  questionType: QuestionType,
}

export interface IOwnProps {
  flashCardStatus: FlashCardStatus,
  questionObject: IQuestionObject,
  onSkip: () => void,
  classes: {
    paper: string,
    boxButtonContainer: string,
    boxQuestionaireContainer: string
  }
}