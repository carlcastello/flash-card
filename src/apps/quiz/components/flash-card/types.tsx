import { IFlashCard } from "../../../commons/types";

export enum FlashCardStatus {
  DEFAULT = "DEFAULT",
  CORRECT = "CORRECT",
  WRONG = "WRONG",
  HINT = "HINT"
}

export interface IOwnProps {
  update: (flashCardStatus: FlashCardStatus) => void,
  next: () => void,
  flashCardObject: IFlashCard,
  classes: {

  }
}

export interface IOwnState {
  flashCardStatus: FlashCardStatus
}