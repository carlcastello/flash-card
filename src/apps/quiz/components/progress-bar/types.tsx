import { FlashcardStatus } from "../flash-card/types";

export interface IOwnProps {
  currentQuestion: number,
  totalQuestion: number,
  flashcardStatus: FlashcardStatus,
  classes: {
    title: string,
    progressBarContainer: string,
    progressBar: string,
  }
}