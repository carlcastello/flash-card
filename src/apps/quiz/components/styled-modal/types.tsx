export interface IOwnProps {
  isOpen: boolean,
  onConfirm?: () => void,
  onIgnore: () => void,

  classes: {
    modalContainer: string,
    modalPaperContainer: string,
    closeButton: string,
  }
}