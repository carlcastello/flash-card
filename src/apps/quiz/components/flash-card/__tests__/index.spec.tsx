import React from "react";
import { shallow } from "enzyme";

import { Flashcard } from "../index"; 
import toJson from "enzyme-to-json";
import AnswerCard from "../components/answer-card";
import ButtonCard from "../components/button-card";
import QuestionCard from "../components/question-card";
import { FlashcardStatus } from "../types";


describe("The Flash Card", () => {

  let updateFunction: any;
  let nextFunction: any;

  let wrapper: any;
  let instance: any;

  beforeEach(() => {
    updateFunction = jest.fn();
    nextFunction = jest.fn();

    wrapper = shallow(
      <Flashcard
        update={updateFunction}
        next={nextFunction}
        question={{
          id: "id",
          question: "question",
          subQuestion: "sub-question",
          hint: "hint",
          answer: "answer",
        }}
        flashcardStatus={FlashcardStatus.DEFAULT}
        classes={{}}/>
    );

    instance = wrapper.instance();
  });

  describe("the render function", () => {
    it("renders", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(QuestionCard)).toHaveLength(1);
      expect(wrapper.find(AnswerCard)).toHaveLength(1);
      expect(wrapper.find(ButtonCard)).toHaveLength(1);
    });
  });

  describe("the onSubmit Function", () => {
    it("calls the updateFunction with FlashcardStatus.CORRECT", () => {
      instance.answerCardRef.current = { state: {answer: "answer"} };

      instance.onSubmit();
      expect(updateFunction).toHaveBeenCalledWith(FlashcardStatus.CORRECT);
    });

    it("calls the updateFunction with FlashcardStatus.WRONG", () => {
      instance.answerCardRef.current = { state: {answer: "potato"} };

      instance.onSubmit();
      expect(updateFunction).toHaveBeenCalledWith(FlashcardStatus.WRONG);
    });
  });

  describe("the onNext Function", () => {
    it("calls the next", () => {
      const setStateFunction: any = jest.fn();

      instance.answerCardRef.current = { setState: setStateFunction };

      instance.onNext();

      expect(nextFunction).toHaveBeenCalledTimes(1);
      expect(setStateFunction).toHaveBeenCalledWith({ answer: "" });
    });
  });

  describe("the onSkip Function", () => {
    it("calls the update function with FlashcardStatus.HINT", () => {
      instance.onSkip();
      expect(updateFunction).toHaveBeenCalledWith(FlashcardStatus.HINT);
    });
  });
});