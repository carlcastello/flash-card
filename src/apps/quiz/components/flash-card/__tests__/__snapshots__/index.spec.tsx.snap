// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Flash Card the render function renders 1`] = `
<Styled(MuiBox)
  pt={2.5}
  px={5}
>
  <form>
    <WithStyles(QuestionCard)
      flashcardStatus="DEFAULT"
      onSkip={[Function]}
      questionObject={
        Object {
          "hint": "hint",
          "question": "question",
          "subQuestion": "sub-question",
        }
      }
    />
    <WithStyles(AnswerCard)
      answer="answer"
      flashcardStatus="DEFAULT"
    />
    <WithStyles(ButtonCard)
      flashcardStatus="DEFAULT"
      next={[Function]}
      submit={[Function]}
    />
  </form>
</Styled(MuiBox)>
`;
